src/main/java/frc/robot/subsystems/drivetrain/SwerveModule.java

```diff
@@ -109,35 +111,49 @@ public SwerveModule(int drivePort, int steerPort, double offsetRads) {
             steerPidController.setPositionPIDWrappingMinInput(0.0);
             steerPidController.setPositionPIDWrappingMaxInput(2 * Math.PI);
         });
-
-        shuffleboardTab = Shuffleboard.getTab("Swerve " + drivePort + " " + steerPort);
-        targetVelEntry = shuffleboardTab.add("Target velocity (mps)", 0.0)
-            .withPosition(0, 0)
-            .withSize(2, 1)
-            .getEntry();
-        currentVelEntry = shuffleboardTab.add("Current velocity (mps)", 0.0)
-            .withPosition(2, 0)
-            .withSize(2, 1)
-            .getEntry();
-        velErrorEntry = shuffleboardTab.add("Velocity error (mps)", 0.0)
-            .withPosition(0, 1)
-            .withSize(5, 3)
-            // .withWidget(BuiltInWidgets.kGraph)
-            .getEntry();
-
-        targetAngleEntry = shuffleboardTab.add("Target angle (degs)", 0.0)
-            .withPosition(6, 0)
-            .withSize(2, 1)
-            .getEntry();
-        currentAngleEntry = shuffleboardTab.add("Current angle (degs)", 0.0)
-            .withPosition(8, 0)
-            .withSize(2, 1)
-            .getEntry();
-        angleErrorEntry = shuffleboardTab.add("Angle error (rads)", 0.0)
-            .withPosition(6, 1)
-            .withSize(5, 3)
-            // .withWidget(BuiltInWidgets.kGraph)
-            .getEntry();
+        
+        this.VERBOSE_SHUFFLEBOARD = VERBOSE_SHUFFLEBOARD;
+
+        if(VERBOSE_SHUFFLEBOARD) {
+            shuffleboardTab = Shuffleboard.getTab("Swerve " + drivePort + " " + steerPort);
+
+            targetVelEntry = shuffleboardTab.add("Target velocity (mps)", 0.0)
+                .withPosition(0, 0)
+                .withSize(2, 1)
+                .getEntry();
+            currentVelEntry = shuffleboardTab.add("Current velocity (mps)", 0.0)
+                .withPosition(2, 0)
+                .withSize(2, 1)
+                .getEntry();
+            velErrorEntry = shuffleboardTab.add("Velocity error (mps)", 0.0)
+                .withPosition(0, 1)
+                .withSize(5, 3)
+                // .withWidget(BuiltInWidgets.kGraph)
+                .getEntry();
+
+            targetAngleEntry = shuffleboardTab.add("Target angle (degs)", 0.0)
+                .withPosition(6, 0)
+                .withSize(2, 1)
+                .getEntry();
+            currentAngleEntry = shuffleboardTab.add("Current angle (degs)", 0.0)
+                .withPosition(8, 0)
+                .withSize(2, 1)
+                .getEntry();
+            angleErrorEntry = shuffleboardTab.add("Angle error (rads)", 0.0)
+                .withPosition(6, 1)
+                .withSize(5, 3)
+                // .withWidget(BuiltInWidgets.kGraph)
+                .getEntry();
+        }
+        else{
+            shuffleboardTab = null;
+            targetVelEntry = null;
```

Comment:
Unless you modify the shuffleboard updates in `setDesiredState()` to have null checks as well, this will cause a runtime exception.